### DISCUSSION TITLE
Is there a way to change the cache directory?

### DISCUSSION THREAD

**Discussion by sclu1034** [clangd/clangd#1109] [Q&A] [answered]
Currently, clangd always creates a directory `.cache/clangd/` at the project root.
Is there a way to configure this location?

Ideally, in a way that I can specify a path outside the project, such as `$XDG_CACHE_HOME/clangd/<project_name>` or `/tmp/clangd_cache/<clangd_pid>`.

**Comment by HighCommander4** [comment, marked as answer]
Currently, clangd always places the index into the same directory as the `compile_commands.json`.

So, a current way to relocate the index from the project root, is to relocate the `compile_commands.json` from the project root and point clangd to it using the `--compile-commands-dir=<directory>` command line argument to clangd.

Customizing the index location independently of the `compile_commands.json` is tracked in https://github.com/clangd/clangd/issues/184.
